File: linear_search.lua
function busqueda(lista, objetivo)
-- Search for a number in a random list of integers. Big-O Complexity: O(n)
  match = false

  for elemento = 0, lista.size(), 1 do
    if lista[elemento] == objetivo then
      match = true
      break
    end
  end

  return match
end

function random_list(size)
  -- Generate a random list of integers. Big-O Complexity: O(n)
  math.randomseed(os.time())
  list = {}

  for i = 0, size do
    list[i] = math.random(0, 100)
  end

  -- added a function to know the size of the array
  list.size = function() return size end
  return list
end

list_int = random_list(20)
ENCONTRADO = busqueda(list_int, 7)
for elemento = 0, list_int.size(), 1 do print(list_int[elemento]) end
if ENCONTRADO == true then print("esta") else print("no esta") end

File: cpp/linear_search.cpp
#include <random>
#include <iostream>
#include <vector>
using namespace std;

bool busqueda(vector<int> lista, int objetivo)
// Search for a number in a random list of integers. Big-O Complexity: O(n)
{
    bool match = false;

    for(int elemento : lista)
    {
        if(elemento == objetivo)
        {
            match = true;
            break;
        }
    }

    return match;
}

int main()
{
    vector<int> list_int;
    srand(time(0));
    for(int i(0); i < 20; i++)
    {
        list_int.push_back((rand() % 100) + 1);
    }

    bool ENCONTRADO = busqueda(list_int, 7);
    for (int elemento : list_int) { cout << elemento << " "; }
    ENCONTRADO ? cout << "\n esta \n": cout << "\n no esta \n";

    return 0;
}
